src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -162,176 +154,453 @@ private function compileRoutes(RouteCollection $routes, $supportsRedirections)
     {
-        $code = '';
-        $prefix = $collection->getPrefix();
-
-        if (!empty($prefix) && '/' !== $prefix) {
-            $code .= sprintf("    %s (0 === strpos(\$pathinfo, %s)) {\n", $ifOrElseIf, var_export($prefix, true));
+        if (!$staticRoutes) {
+            return '';
         }
+        $code = $default = '';
+        $checkTrailingSlash = false;
 
-        $ifOrElseIf = 'if';
+        foreach ($staticRoutes as $url => $routes) {
+            if (1 === count($routes)) {
+                foreach ($routes as $name => list($hasTrailingSlash, $route)) {
+                }
 
-        foreach ($collection->getItems() as $route) {
-            if ($route instanceof StaticPrefixCollection) {
-                $code .= $this->compileStaticPrefixRoutes($route, $supportsRedirections, $ifOrElseIf);
-                $ifOrElseIf = 'elseif';
-            } else {
-                $code .= $this->compileRoute($route[1]->getRoute(), $route[1]->getName(), $supportsRedirections, $prefix)."\n";
-                $ifOrElseIf = 'if';
+                if (!$route->getCondition()) {
+                    if (!$supportsRedirections && $route->getSchemes()) {
+                        throw new \LogicException('The "schemes" requirement is only supported for URL matchers that implement RedirectableUrlMatcherInterface.');
+                    }
+                    $checkTrailingSlash = $checkTrailingSlash || $hasTrailingSlash;
+                    $default .= sprintf(
+                        "%s => array(%s, %s, %s, %s),\n",
+                        self::export($url),
+                        self::export(array('_route' => $name) + $route->getDefaults()),
+                        self::export(!$route->compile()->getHostVariables() ? $route->getHost() : $route->compile()->getHostRegex() ?: null),
+                        self::export(array_flip($route->getMethods()) ?: null),
+                        self::export(array_flip($route->getSchemes()) ?: null).($hasTrailingSlash ? ', true' : '')
+                    );
+                    continue;
+                }
             }
+
+            $code .= sprintf("        case %s:\n", self::export($url));
+            foreach ($routes as $name => list($hasTrailingSlash, $route)) {
+                $code .= $this->compileRoute($route, $name, $supportsRedirections, $hasTrailingSlash);
+            }
+            $code .= "            break;\n";
         }
 
-        if (!empty($prefix) && '/' !== $prefix) {
-            $code .= "    }\n\n";
-            // apply extra indention at each line (except empty ones)
-            $code = preg_replace('/^.{2,}$/m', '    $0', $code);
+        $matchedPathinfo = $supportsRedirections ? '$trimmedPathinfo' : '$pathinfo';
+
+        if ($default) {
+            $code .= <<<EOF
+        default:
+            \$routes = array(
+{$this->indent($default, 4)}            );
```

Comment:
This array can contain severals static routes. With a big amount of route, this array and the `$regexList` and `$matchedPathinfo` could be significant.

Isn't it possible to define them as static array and benefit of OPCode cache ? (But won't work on 3.4 with php55 requirement...) Like you've done with the php array adapter ;)

Anyway great job @nicolas-grekas !

Reply:
A static array can be inline, opcache will work with them. So nothing to change here to benefit from shared memory.